### PR TITLE
[batch] Billing query by interval slowness fix

### PR DESCRIPTION
**Pull request by jigold**
I think we want to see deleted batches in this UI page. The point is to see all costs.

### CODE REVIEW

**Comment by jigold** [comment]
If not, we need to fix the query to not be slow.

File: batch/batch/front_end/front_end.py [hunk 1]
```diff
@@ -1378,8 +1378,7 @@ async def parse_error(msg):
 LEFT JOIN billing_projects
   ON billing_projects.name = batches.billing_project
 WHERE `time_completed` >= %s AND
-  `time_completed` <= %s AND
-  billing_projects.`status` != 'deleted'
+  `time_completed` <= %s
```
Comment (danking) [inline batch/batch/front_end/front_end.py:1381]: The billing_project index includes deleted status. I tested this just now on querying all of November, it takes 3.75 seconds without the deleted ones and 6.91 to with no status filter.

Billing project deletion is used by the tests to remove test billing projects. An open billing project doesn't have a delete button, just a close button, so I don't think it's likely to happen by accident.

```
mysql> explain SELECT   billing_project,   `user`,   CAST(SUM(IF(format_version < 3, batches.msec_mcpu, 0)) AS SIGNED) as msec_mcpu,   SUM(IF(format_version >= 3, `usage` * rate, NULL)) as cost FROM batches LEFT JOIN aggregated_batch_resources   ON aggregated_batch_resources.batch_id = batches.id LEFT JOIN resources
  ON resources.resource = aggregated_batch_resources.resource  LEFT JOIN billing_projects    ON billing_projects.name = batches.billing_project  WHERE `time_completed` >= 1606861283280 AND `time_completed` <= 1607638653000 and billing_projects.status != 'deleted' GROUP BY billing_project, `user`;
+----+-------------+----------------------------+------------+--------+------------------------------------------------------+------------------------+---------+-------------------------------------------+------+----------+--------------------------------------------------------+
| id | select_type | table                      | partitions | type   | possible_keys                                        | key                    | key_len | ref                                       | rows | filtered | Extra                                                  |
+----+-------------+----------------------------+------------+--------+------------------------------------------------------+------------------------+---------+-------------------------------------------+------+----------+--------------------------------------------------------+
|  1 | SIMPLE      | batches                    | NULL       | range  | batches_time_completed,batches_billing_project_state | batches_time_completed | 9       | NULL                                      | 5759 |   100.00 | Using index condition; Using temporary; Using filesort |
|  1 | SIMPLE      | billing_projects           | NULL       | eq_ref | PRIMARY,billing_project_status                       | PRIMARY                | 302     | batch.batches.billing_project             |    1 |    66.67 | Using where                                            |
|  1 | SIMPLE      | aggregated_batch_resources | NULL       | ref    | PRIMARY                                              | PRIMARY                | 8       | batch.batches.id                          |   36 |   100.00 | NULL                                                   |
|  1 | SIMPLE      | resources                  | NULL       | eq_ref | PRIMARY                                              | PRIMARY                | 302     | batch.aggregated_batch_resources.resource |    1 |   100.00 | NULL                                                   |
+----+-------------+----------------------------+------------+--------+------------------------------------------------------+------------------------+---------+-------------------------------------------+------+----------+--------------------------------------------------------+
```

**Review by danking — CHANGES_REQUESTED** [review]
The issue here is the sheer magnitude of batches in `ci` and `test`. For example, I ran these queries back-to-back a few times to eliminate any caching differences. I save >95% of the time by not aggregating over ci and test batches.

Why do we need to fix the query? Isn't this just used by Whitney to compute billing reports once a month?

```
mysql> SELECT
    ->   billing_project,
    ->   `user`,
    ->   CAST(SUM(IF(format_version < 3, batches.msec_mcpu, 0)) AS SIGNED) as msec_mcpu,
    ->   SUM(IF(format_version >= 3, `usage` * rate, NULL)) as cost
    -> FROM batches
    -> LEFT JOIN aggregated_batch_resources
    ->   ON aggregated_batch_resources.batch_id = batches.id
    -> LEFT JOIN resources
    ->   ON resources.resource = aggregated_batch_resources.resource
    -> LEFT JOIN billing_projects
    ->   ON billing_projects.name = batches.billing_project
    -> WHERE `time_completed` >= 1606861283280 AND
    ->   `time_completed` <= 1607638653000 AND
    ->   billing_projects.`status` != 'deleted' AND
    ->   billing_project != "ci" AND
    ->   billing_project != "test"
    -> GROUP BY billing_project, `user`;
+------------------------+-----------+-----------+-------------------------+
| billing_project        | user      | msec_mcpu | cost                    |
+------------------------+-----------+-----------+-------------------------+
| benchmark              | benchmark |         0 |    0.014514134476718753 |
| broad-mpg-gnomad       | mwilson   |         0 | 0.000006647671923720436 |
| daly-neale-sczmeta     | fgulamal  |         0 |    0.009850052299461938 |
| daly-neale-sczmeta     | tsingh    |         0 |    0.002160176529251274 |
| hail                   | dking     |         0 |     0.03902524028628604 |
| hail                   | johnc     |         0 |       8.774393078244739 |
| tgg-rare-disease       | weisburd  |         0 |      129.65639737090416 |
| ukb-pharma-browser-dev | msolomon  |         0 |     0.18594499876264362 |
+------------------------+-----------+-----------+-------------------------+
8 rows in set (0.08 sec)

mysql> SELECT
    ->   billing_project,
    ->   `user`,
    ->   CAST(SUM(IF(format_version < 3, batches.msec_mcpu, 0)) AS SIGNED) as msec_mcpu,
    ->   SUM(IF(format_version >= 3, `usage` * rate, NULL)) as cost
    -> FROM batches
    -> LEFT JOIN aggregated_batch_resources
    ->   ON aggregated_batch_resources.batch_id = batches.id
    -> LEFT JOIN resources
    ->   ON resources.resource = aggregated_batch_resources.resource
    -> LEFT JOIN billing_projects
    ->   ON billing_projects.name = batches.billing_project
    -> WHERE `time_completed` >= 1606861283280 AND
    ->   `time_completed` <= 1607638653000 AND
    ->   billing_projects.`status` != 'deleted'
    -> GROUP BY billing_project, `user`;

+------------------------+-----------+-----------+-------------------------+
| billing_project        | user      | msec_mcpu | cost                    |
+------------------------+-----------+-----------+-------------------------+
| benchmark              | benchmark |         0 |    0.014514134476718753 |
| broad-mpg-gnomad       | mwilson   |         0 | 0.000006647671923720436 |
| ci                     | ci        |         0 |       50.76884205937543 |
| daly-neale-sczmeta     | fgulamal  |         0 |    0.009850052299461938 |
| daly-neale-sczmeta     | tsingh    |         0 |    0.002160176529251274 |
| hail                   | dking     |         0 |     0.03902524028628604 |
| hail                   | johnc     |         0 |       8.774393078244739 |
| test                   | test      |         0 |      0.7429831609790919 |
| test                   | test-dev  |         0 |    0.008862601871767828 |
| tgg-rare-disease       | weisburd  |         0 |      129.65639737090416 |
| ukb-pharma-browser-dev | msolomon  |         0 |     0.18594499876264362 |
+------------------------+-----------+-----------+-------------------------+
11 rows in set (1.23 sec)
```

**Comment by danking** [comment]
A little more context, this is the number of batch resources per project, per user. Ci and test are an order of magnitude more than Ben W. who is two orders of magnitude above his nearest competitors.
```
+------------------------+-----------+----------+
| billing_project        | user      | count(*) |
+------------------------+-----------+----------+
| benchmark              | benchmark |     1595 |
| broad-mpg-gnomad       | mwilson   |       17 |
| ci                     | ci        |   328597 |
| daly-neale-sczmeta     | fgulamal  |      292 |
| daly-neale-sczmeta     | tsingh    |       71 |
| hail                   | dking     |     1610 |
| hail                   | johnc     |     5932 |
| test                   | test      |   126717 |
| test                   | test-dev  |     2347 |
| tgg-rare-disease       | weisburd  |    10534 |
| ukb-pharma-browser-dev | msolomon  |      161 |
+------------------------+-----------+----------+
```

**Pull request closed without merging**